target_chains/aptos/contracts/sources/pyth.move

```diff
@@ -395,6 +394,119 @@ module pyth::pyth {
+            let price_info = vector::borrow(price_infos, i);
+            let price_feed = price_info::get_price_feed(price_info);
+            let price: price::Price = price_feed::get_price(price_feed);
+            let timestamp = price::get_timestamp(&price);
+            if (timestamp >= min_publish_time && timestamp <= max_publish_time) {
+                let price_id: &price_identifier::PriceIdentifier = price_feed::get_price_identifier(price_feed);
+                let price_id_bytes = price_identifier::get_bytes(price_id);
+                vector::push_back(&mut parsed_price_ids, price_id_bytes);
+                vector::push_back(&mut parsed_price_feeds, *price_feed);
+            };
+            i = i + 1;
+        };
+
+        // ensure all requested price IDs have corresponding valid updates
+        let k = 0;
+        while (k < vector::length(&price_ids)) {
+            let requested_price_id = vector::borrow(&price_ids, k);
+            let found = false;
+
+            let j = 0;
+            while (j < vector::length(&parsed_price_ids)) {
+                let parsed_price_id = vector::borrow(&parsed_price_ids, j);
+                if (requested_price_id == parsed_price_id) {
+                    found = true;
+                };
+                j = j + 1;
+            };
+
+            if (!found) {
+                abort error::unknown_price_feed() // or replace with more suitable error 
+            };
+            k = k + 1;
+        };
+
+        return parsed_price_feeds
+    }
+
+    // parse a single VAA and return vector of price feeds 
+    fun parse_price_feed_updates_single_vaa (
+        vaa: vector<u8>,
+        price_ids: vector<vector<u8>>,
+        min_publish_time: u64, 
+        max_publish_time: u64
+    ): vector<price_feed::PriceFeed> {
+        let cur = cursor::init(vaa);
+        let header: u64 = deserialize::deserialize_u32(&mut cur);
+        if (header == PYTHNET_ACCUMULATOR_UPDATE_MAGIC) {
+            let price_infos = parse_and_verify_accumulator_message(&mut cur);
+            cursor::rest(cur);
+            return parse_and_validate_price_feeds(&price_infos, price_ids, min_publish_time, max_publish_time)
+        } else {
+            let vaa = vaa::parse_and_verify(vaa);
+            verify_data_source(&vaa);
+            let price_infos = batch_price_attestation::destroy(batch_price_attestation::deserialize(vaa::destroy(vaa)));
+            cursor::rest(cur);
+            return parse_and_validate_price_feeds(&price_infos, price_ids, min_publish_time, max_publish_time)
+        }
+    }
+
+    public fun parse_price_feed_updates(
+        update_data: vector<vector<u8>>,
+        price_ids: vector<vector<u8>>,
+        min_publish_time: u64,
+        max_publish_time: u64,
+        fee: Coin<AptosCoin>
+    ): vector<price_feed::PriceFeed> {
+        // validate and deposit the update fee 
+        let update_fee = get_update_fee(&update_data);
```

Comment:
this will run all the verification again. You can assert the fees were enough at the end of the function using the length of the price_feeds. Same as what we have done on `update_price_feeds`